File: 프로그래머스/level1/신규 아이디 추천.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string first(string id){
    for(int i=0; i<id.length(); i++){
        if(id[i]>='A'&&id[i]<='Z'){
            id[i] += 32;
        }
    }
    string tmp = "";
    for(int i=0; i<id.length(); i++){
        if(id[i]=='.'||id[i]=='-'||id[i]=='_'||(id[i]>='0'&&id[i]<='9')||(id[i]>='a'&&id[i]<='z'))  tmp += id[i];
    }
    string tmp1 = "";
    tmp1 += tmp[0];
    for(int i=1; i<tmp.length(); i++){
        if(tmp[i-1]=='.'&&tmp[i]=='.')  continue;
        tmp1 += tmp[i];
    }
    if(tmp1[0]=='.'){
        tmp1 = tmp1.substr(1);
    }
    if(tmp1[tmp1.length()-1]=='.')  tmp1.pop_back();
    if(tmp1.compare("")==0){
        tmp1 = "a";
    }
    if(tmp1.length()>=16){
        tmp1 = tmp1.substr(0,15);
    }
    if(tmp1[tmp1.length()-1]=='.')  tmp1.pop_back();
    if(tmp1.length()<=2){
        while(tmp1.length()<3){
            tmp1.push_back(tmp1[tmp1.length()-1]);
        }
    }
    return tmp1;
}
string solution(string new_id) {
    string answer = "";
    answer = first(new_id);
    
    return answer;
}

File: 프로그래머스/level1/신규 아이디 추천.js
const step = (str) => {
    str = str.toLowerCase();
    let ans = "";
    for(let i=0; i<str.length; i++){
        if((str[i]>='a'&&str[i]<='z')||(str[i]>='0'&&str[i]<='9')||str[i]==="-"||str[i]==="_"||str[i]==="."){
            ans += str[i];
        }
    }
    let ans2 = ans[0];
    for(let i=1; i<ans.length; i++){
        if(ans2[ans2.length-1]==='.'&&ans[i]==='.')   continue;
        ans2 += ans[i];
    }
    if(ans2[0]==='.'){
        ans2 = ans2.slice(1);
    }
    if(ans2[ans2.length-1]==='.'){
        ans2 = ans2.slice(0, ans2.length-1);
    }
    if(ans2.length===0){
        ans2 += 'a';
    }
    if(ans2.length>=16){
        ans2 = ans2.slice(0,15);
    }
    if(ans2[ans2.length-1]==='.'){
        ans2 = ans2.slice(0, ans2.length-1);
    }
    if(ans2.length<=2){
        while(ans2.length!==3){
            ans2 += ans2[ans2.length-1];
        }
    }
    return ans2;
}

function solution(new_id) {
    var answer = '';
    answer = step(new_id);
    return answer;
}